usr/src/cmd/zdb/zdb.c

```diff
@@ -3709,8 +3710,17 @@ main(int argc, char **argv)
 		}
 	}
 
+	if (strpbrk(target, "/@") != NULL) {
+		size_t targetlen;
+
+		target_is_spa = B_FALSE;
+		targetlen = strlen(target);
+		if (targetlen && target[targetlen - 1] == '/')
+			target[targetlen - 1] = '\0';
```

Comment:
Around here would be a great place for a comment explaining why we do this.  Presumably because later code will get confused by a trailing slash, but we allow it specifically for the "pool/" case.